hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveSessionImpl.java

```diff
@@ -359,8 +360,20 @@ else if ( isPersistentAttributeInterceptable( association ) ) {
 
 	@Override
 	public <E, T> CompletionStage<T> reactiveFetch(E entity, Attribute<E, T> field) {
-		return ( (ReactiveEntityPersister) getEntityPersister( null, entity ) )
-				.reactiveInitializeLazyProperty( field, entity, this );
+		final ReactiveEntityPersister entityPersister = (ReactiveEntityPersister) getEntityPersister( null, entity );
+		LazyAttributeLoadingInterceptor lazyAttributeLoadingInterceptor = entityPersister.getBytecodeEnhancementMetadata()
+				.extractInterceptor( entity );
+		final String attributeName = field.getName();
+		if ( !lazyAttributeLoadingInterceptor.isAttributeLoaded( attributeName ) ) {
+			return ( (CompletionStage<T>) lazyAttributeLoadingInterceptor.fetchAttribute( entity, field.getName() ) )
+					.thenApply( value -> {
+						lazyAttributeLoadingInterceptor.attributeInitialized( attributeName );
+						return value;
+					} );
+		}
+		else {
+			return completedFuture( (T) entityPersister.getPropertyValue( entity, attributeName ) );
```

Comment:
If I understand correctly, this change will prevent an extra query. Could you make sure we test for it?
Otherwise, future changes might remove the improvement.